test/modules/aws-codedeploy-integration.ts

```diff
@@ -0,0 +1,618 @@
+
+import * as iasql from '../../src/services/iasql';
+import {
+  execComposeDown,
+  execComposeUp,
+  finish,
+  getPrefix,
+  runApply,
+  runInstall,
+  runQuery,
+  runSync,
+  runUninstall,
+} from '../helpers';
+
+const prefix = getPrefix();
+const dbAlias = 'codedeploytest';
+const applicationName = `${prefix}${dbAlias}application`;
+const applicationNameForDeployment = `${prefix}${dbAlias}applicationForDeployment`;
+const deploymentGroupName = `${prefix}${dbAlias}deployment_group`;
+const ubuntuAmiId =
+  'resolve:ssm:/aws/service/canonical/ubuntu/server/20.04/stable/current/amd64/hvm/ebs-gp2/ami-id';
+
+const region = process.env.AWS_REGION ?? '';
+const accessKeyId = process.env.AWS_ACCESS_KEY_ID ?? '';
+const secretAccessKey = process.env.AWS_SECRET_ACCESS_KEY ?? '';
+const ec2client = new EC2({
+  credentials: {
+    accessKeyId,
+    secretAccessKey,
+  },
+  region,
+});
+
+const getAvailabilityZones = async () => {
+  return await ec2client.describeAvailabilityZones({
+    Filters: [
+      {
+        Name: 'region-name',
+        Values: [region],
+      },
+    ],
+  });
+};
+
+const getInstanceTypeOffering = async (availabilityZones: string[]) => {
+  return await ec2client.describeInstanceTypeOfferings({
+    LocationType: 'availability-zone',
+    Filters: [
+      {
+        Name: 'location',
+        Values: availabilityZones,
+      },
+      {
+        Name: 'instance-type',
+        Values: ['t2.micro', 't3.micro'],
+      },
+    ],
+  });
+};
+
+const roleName = `${prefix}-codedeploy-${region}`;
+const ec2RoleName = `${prefix}-codedeploy-ec2-${region}`;
+
+const codedeployRolePolicy = JSON.stringify({
+  Version: '2012-10-17',
+  Statement: [
+    {
+      Sid: '',
+      Effect: 'Allow',
+      Principal: {
+        Service: 'codedeploy.amazonaws.com',
+      },
+      Action: 'sts:AssumeRole',
+    },
+  ],
+});
+const codedeployPolicyArn = 'arn:aws:iam::aws:policy/AWSCodeDeployFullAccess';
+const deployEC2PolicyArn = 'arn:aws:iam::aws:policy/AmazonEC2FullAccess';
+
+const ec2RolePolicy = JSON.stringify({
+  Version: '2012-10-17',
+  Statement: [
+    {
+      Sid: '',
+      Effect: 'Allow',
+      Principal: {
+        Service: 'ec2.amazonaws.com',
+      },
+      Action: 'sts:AssumeRole',
+    },
+  ],
+});
+const ssmPolicyArn = 'arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore';
+const instanceTag = `${prefix}-codedeploy-vm`;
+
+const ec2FilterTags = JSON.stringify([
+  {
+    Type: 'KEY_AND_VALUE',
+    Key: 'name',
+    Value: `${instanceTag}`,
+  },
+]);
+
+const revisionLocationv0 = JSON.stringify({
+  revisionType: 'GitHub',
+  gitHubLocation: {
+    repository: 'iasql/iasql-codedeploy-example',
+    commitId: 'cf6aa63cbd2502a5d1064363c2af5c56cc2107cc',
+  },
+});
+
+const revisionLocationv1 = JSON.stringify({
+  revisionType: 'GitHub',
+  gitHubLocation: {
+    repository: 'iasql/iasql-codedeploy-example',
+    commitId: '165582e107955f0b114a9d9d74cd2e4f198454a7',
+  },
+});
```

Comment:
how is authentication working atm?

Reply:
no authentication, we can just use public repos